### PR TITLE
Run tests on PHP 8.3 and update test suite

### PR DESCRIPTION
**Pull request by yadaiio**
This pull request builds on top of #108 #107 and #105.

References: https://github.com/clue/reactphp-csv/pull/33, https://github.com/reactphp/socket/pull/300 and https://github.com/reactphp/socket/pull/299.

After this pull request is merged, we can continue with #112

### CODE REVIEW

File: phpunit.xml.dist [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
-<!-- PHPUnit configuration file with new format for PHPUnit 9.3+ -->
-<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" 
-         xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/9.3/phpunit.xsd"
+<!-- PHPUnit configuration file with new format for PHPUnit 9.6+ -->
+<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/9.6/phpunit.xsd"
          bootstrap="vendor/autoload.php"
-         colors="true"
          cacheResult="false">
+         colors="true"
```
Comment (SimonFrings) [inline phpunit.xml.dist]:
```suggestion
         cacheResult="false"
         colors="true">
```

The closing bracket here is in the wrong place, should be a quick fix.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -37,13 +38,16 @@ jobs:
 
   PHPUnit-hhvm:
     name: PHPUnit (HHVM)
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-22.04
```
Comment (SimonFrings) [inline .github/workflows/ci.yml:41]: Using `ubuntu-22.04` makes perfect sense :+1: 

It appears that we're still using the old `ubuntu-20.04` in our PHPUnit job within the `ci.yml` file. I believe this should also be updated to `ubuntu-22.04`, just like you did for the PHPUnit-hhvm job.

**Review by SimonFrings — CHANGES_REQUESTED** [review]
Thanks for filing this pull request :+1: 

I added a few remarks below, should both be quick fixes.

I also noticed that the test suite currently reports unhandled promise rejections in this project (see https://github.com/reactphp/promise/pull/248 and https://github.com/orgs/reactphp/discussions/525 for reference), but I think this is something for a follow-up PR. I will look into this.

**Comment by yadaiio** [comment]
Tysm @SimonFrings for seeing both mistakes! Helps me a lot for the future. 🥳 Should be alright now!

**Review by SimonFrings — APPROVED** [review]
Thanks for applying the changes, looking good :+1:

File: phpunit.xml.dist [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
-<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" 
-         xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/9.3/phpunit.xsd"
+<!-- PHPUnit configuration file with new format for PHPUnit 9.6+ -->
+<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/9.6/phpunit.xsd"
          bootstrap="vendor/autoload.php"
-         colors="true"
-         cacheResult="false">
+         cacheResult="false"
+         colors="true">
```
Comment (clue) [inline phpunit.xml.dist:8]: Should probably also include `convertDeprecationsToExceptions="true"` to fail for any deprecations as done in other projects like https://github.com/reactphp/socket/pull/277.

**Review by clue — CHANGES_REQUESTED** [review]

File: phpunit.xml.dist (hunk 3 above)
Comment (yadaiio) [reply phpunit.xml.dist:8]: whoops, I did change it indeed, but I forgot to resolve this conversation. Thanks @clue 👍

**Review by clue — APPROVED** [review]
@yadaiio Thanks for filing this PR!

I've updated the PR to no longer report any deprecations as reported in #114. This PR now no longer reports any test errors and focuses on adding PHP 8.3 only.

I'll file separate follow-up PRs to address the deprecations reported in #114 and the unhandled promise rejections reported in #115.

**Comment by yadaiio** [comment]
Thank you @clue for the update 👍

**Pull request merged**